modules/nvidia_plugin/src/cuda_plugin.cpp

```diff
@@ -270,11 +271,23 @@ InferenceEngine::Parameter Plugin::GetMetric(const std::string& name,
         //        all_properties.insert(all_properties.end(), configKeys.begin(), configKeys.end());
         //        return all_properties;
     } else if (METRIC_KEY(AVAILABLE_DEVICES) == name) {
-        // TODO: fill list of available devices
-        std::vector<std::string> availableDevices = {""};
-        IE_SET_METRIC_RETURN(AVAILABLE_DEVICES, availableDevices);
+        std::vector<std::string> availableDevices = { };
+        for (size_t i = 0; i < CUDA::Device::count(); ++i) {
+            availableDevices.push_back(fmt::format("{}.{}", _pluginName, i));
+        }
+        return decltype(ov::available_devices)::value_type {availableDevices};
+    } else if (ov::device::uuid == name) {
+        const std::string deviceId = _cfg.Get(CONFIG_KEY(DEVICE_ID));
+        CUDA::Device device{std::stoi(deviceId)};
+        const auto& props = device.props();
+        ov::device::UUID uuid = {};
+        std::copy(std::begin(props.uuid.bytes), std::end(props.uuid.bytes), std::begin(uuid.uuid));
+        return decltype(ov::device::uuid)::value_type {uuid};
     } else if (METRIC_KEY(FULL_DEVICE_NAME) == name) {
-        std::string name = getCudaAttribute<Plugin::cuda_attribute::name, std::string>();
+        const std::string deviceId = _cfg.Get(CONFIG_KEY(DEVICE_ID));
+        CUDA::Device device{std::stoi(deviceId)};
+        const auto& props = device.props();
+        const std::string name = props.name;
```

Comment:
can we add tests? We can instantiate common tests from main repo

Reply:
@ilya-lavrenov 
Yes, I've already tired to find tests for AVAILABLE_DEVICES and for UUID ...
Can you share a link from other plugin for such test ?

Reply:
Done.
Added test for DEVICE_UUID.
Tests for AVAILABLE_DEVICES was already in test suite